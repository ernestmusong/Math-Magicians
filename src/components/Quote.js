import React, { useState, useEffect } from 'react';
import { FaCircleNotch, FaQuoteLeft } from 'react-icons/fa';
import styles from '../styles/Quotes.module.css';

function Quote() {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  const apiKey = '3s3I81LpJxpp0MD46Yqweg==utIo6B2WhXAtvPcj';
  const category = 'happiness';

  useEffect(() => {
    const fetchQuote = async () => {
      setIsLoading(true);
      try {
        const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
          method: 'GET',
          headers: {
            'X-Api-Key': apiKey,
            'Content-Type': 'application/json',
          },
        });
        const json = await res.json();
        setData(json);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchQuote();
  }, [setData, setIsLoading]);

  if (hasError) {
    return (
      <div className={styles.wrap}>
        <p className={styles.error}>Something went wrong!</p>
      </div>
    );
  }
  if (isLoading) {
    return (
      <div className={styles.wrap}>
        <div className={styles.spinner}><FaCircleNotch size={70} /></div>
      </div>
    );
  }

  return (
    <section className={styles.wrap}>
      <div id={styles.box}>
        {data.map((item) => (
          <blockquote key={data.indexOf(item)} className={styles.blockquote}>
            <p className={styles.quote}>
              <FaQuoteLeft size={42} />
              <span id={styles.text}>{item.quote}</span>
            </p>
            <footer>
              <span className={styles.author} style={{ marginRight: '5px' }}>-</span>
              <span className={styles.author}>{item.author}</span>
            </footer>
          </blockquote>
        ))}
      </div>
    </section>
  );
}

export default Quote;
