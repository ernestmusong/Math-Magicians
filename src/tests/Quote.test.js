import { render, act } from '@testing-library/react';
import Quote from '../components/Quote';

global.fetch = require('jest-fetch-mock');

beforeEach(() => {
  fetch.resetMocks();
  fetch.mockResponse(JSON.stringify([{ quote: 'Everything is not ok', author: 'Musong' }]));
});

it('quote renders correctly', async () => {
  const quote = await act(async () => render(<Quote />));

  expect(quote).toMatchSnapshot();
});
