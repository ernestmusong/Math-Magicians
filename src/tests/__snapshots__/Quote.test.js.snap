// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="wrap"
      >
        <div
          id="box"
        >
          <blockquote
            class="blockquote"
          >
            <p
              class="quote"
            >
              <svg
                fill="currentColor"
                height="42"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="42"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
                />
              </svg>
              <span
                id="text"
              >
                Everything is not ok
              </span>
            </p>
            <footer>
              <span
                class="author"
                style="margin-right: 5px;"
              >
                -
              </span>
              <span
                class="author"
              >
                Musong
              </span>
            </footer>
          </blockquote>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="wrap"
    >
      <div
        id="box"
      >
        <blockquote
          class="blockquote"
        >
          <p
            class="quote"
          >
            <svg
              fill="currentColor"
              height="42"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="42"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
              />
            </svg>
            <span
              id="text"
            >
              Everything is not ok
            </span>
          </p>
          <footer>
            <span
              class="author"
              style="margin-right: 5px;"
            >
              -
            </span>
            <span
              class="author"
            >
              Musong
            </span>
          </footer>
        </blockquote>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
